tests/cover/test_settings.py

```diff
@@ -187,6 +187,33 @@ def test_deprecate_uses_default():
             note_deprecation('Hi')
 
 
+def test_cannot_set_deprecated_settings_with_strict():
+    with pytest.raises(DeprecationWarning):
```

Comment:
We raise a `HypothesisDeprecationWarning`, not a `DeprecationWarning`. No reason not to test the specific exception.